packages/crypto/src/crypto.ts

```diff
@@ -0,0 +1,498 @@
+    publicKeyUncompressed,
+  };
+};
+/**
+ * Convert a hexadecimal string to a Uint8Array.
+ *
+ * @param {string} hexString - The hexadecimal string.
+ * @returns {Uint8Array} - The Uint8Array representation.
+ * @throws {Error} - If the hexadecimal string is invalid.
+ */
+export const uint8arrayFromHexString = (hexString: string): Uint8Array => {
+  const hexRegex = /^[0-9A-Fa-f]+$/;
+  if (!hexString || hexString.length % 2 != 0 || !hexRegex.test(hexString)) {
+    throw new Error(
+      `cannot create uint8array from invalid hex string: "${hexString}"`
+    );
+  }
+  return new Uint8Array(
+    hexString!.match(/../g)!.map((h: string) => parseInt(h, 16))
+  );
+};
+
+/**
+ * Convert a Uint8Array to a hexadecimal string.
+ *  @param {any} array - A Uint8Array (type is any to overcome a typescript error)
+ */
+export const uint8arrayToHexString = (array: any) => {
+  return [...array].map((x) => x.toString(16).padStart(2, "0")).join("");
+};
+
+/**
+ * Build labeled Initial Key Material (IKM).
+ *
+ * @param {Uint8Array} label - The label to use.
+ * @param {Uint8Array} ikm - The input key material.
+ * @param {Uint8Array} suite_id - The suite identifier.
+ * @returns {Uint8Array} - The labeled IKM.
+ */
+const buildLabeledIkm = (
+  label: Uint8Array,
+  ikm: Uint8Array,
+  suite_id: Uint8Array
+): Uint8Array => {
+  const combinedLength =
+    HPKE_VERSION.length + suite_id.length + label.length + ikm.length;
+  const ret = new Uint8Array(combinedLength);
+  let offset = 0;
+
+  ret.set(HPKE_VERSION, offset);
+  offset += HPKE_VERSION.length;
+
+  ret.set(suite_id, offset);
+  offset += suite_id.length;
+
+  ret.set(label, offset);
+  offset += label.length;
+
+  ret.set(ikm, offset);
+
+  return ret;
+};
+
+/**
+ * Build labeled info for HKDF operations.
+ *
+ * @param {Uint8Array} label - The label to use.
+ * @param {Uint8Array} info - Additional information.
+ * @param {Uint8Array} suite_id - The suite identifier.
+ * @param {number} len - The output length.
+ * @returns {Uint8Array} - The labeled info.
+ */
+const buildLabeledInfo = (
+  label: Uint8Array,
+  info: Uint8Array,
+  suite_id: Uint8Array,
+  len: number
+): Uint8Array => {
+  const ret = new Uint8Array(
+    9 + suite_id.byteLength + label.byteLength + info.byteLength
+  );
+  ret.set(new Uint8Array([0, len]), 0);
+  ret.set(HPKE_VERSION, 2);
+  ret.set(suite_id, 9);
+  ret.set(label, 9 + suite_id.byteLength);
+  ret.set(info, 9 + suite_id.byteLength + label.byteLength);
+  return ret;
+};
+
+/**
+ * Convert a BigInt to a hexadecimal string of a specific length.
+ */
+const bigIntToHex = (num: bigint, length: number): string => {
+  const hexString = num.toString(16);
+  if (hexString.length > length) {
+    throw new Error(
+      `number cannot fit in a hex string of ${length} characters`
+    );
+  }
+  return hexString.padStart(length, "0");
+};
+
+/**
+ * Uncompress a raw public key.
+ */
+
+const uncompressRawPublicKey = (rawPublicKey: Uint8Array): Uint8Array => {
```

Comment:
This should probably not be part of our public functions, or we if we want to expose it we need to make sure it reflects the fact that it's P256-specific.